### PR TITLE
Media: Enable Image Editor in Development

### LINKED ISSUE #6639: Image Editor: remove image editor prototype flow
Let's remove the "Add and Edit Image" button, and the extra dropzone we added during the prototype stages.

db3124f0-1234-11e6-9219-5954adf1c8ff
- Start Calypso with `ENABLE_FEATURES=post-editor/image-editor make run` and navigate to
http://calypso.localhost:3000/post
- Select a site
- Click on the media button

### PR DESCRIPTION
**Pull request by gwwar**
This PR fixes #6639 and #6640 and enables the image editor for development only. This removes the temporary workflow that was needed because of cross-origin issues in #5099. 

Note that this PR just enables an example flow, and we'll want to tackle #7375, #7376, #6642 in separate PRs.

## Testing Instructions

1. Navigate to http://calypso.localhost:3000/post
2. Select a site
3. Add an image
4. Select the image
screen shot 2016-08-09 at 2 12 35 pm
5. Click "Edit", then click on "Edit Image" in the top right corner of the detail view
screen shot 2016-08-09 at 2 12 43 pm
6. The image editor should load. Make some edits and click "Done"
screen shot 2016-08-09 at 2 12 58 pm
7. If changes were made, we should go back to the list view, and see a new edited copy of the image being uploaded.
screen shot 2016-08-09 at 2 13 07 pm

cc @robobot3000 @rralian @aduth @lamosty @artpi @retrofox 

Test live: https://calypso.live/?branch=update/image-editor-prototype

### CODE REVIEW

File: client/post-editor/media-modal/image-editor/image-editor-canvas.jsx [hunk 1]
```diff
@@ -66,7 +66,18 @@ const MediaModalImageEditorCanvas = React.createClass( {
...
 		}
 
-		this.initImage( this.props.src );
+		this.getImage( this.props.src );
+	},
+
+	getImage( url ) {
+		const req = new XMLHttpRequest();
+		req.open( 'GET', url, true );
+		req.responseType = "arraybuffer";
```
Comment (aduth) [inline client/post-editor/media-modal/image-editor/image-editor-canvas.jsx]: Single quotes ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#strings))

File: client/post-editor/media-modal/image-editor/index.jsx [hunk 2]
```diff
@@ -49,30 +48,24 @@ const MediaModalImageEditor = React.createClass( {
 			fileName = 'default',
 			mimeType = 'image/png';
 
-		if ( this.props.items && this.props.items[ this.props.selectedIndex ] ) {
-			src = MediaUtils.url( this.props.items[ this.props.selectedIndex ], {
+		const media = this.props.items[ this.props.selectedIndex ];
+
+		if ( this.props.items && media ) {
```
Comment (aduth) [inline client/post-editor/media-modal/image-editor/index.jsx]: `this.props.items` condition redundant at this point if you're already accessing it in the line above?

File: client/post-editor/media-modal/image-editor/image-editor-canvas.jsx (hunk 1 above)
Comment (gwwar) [inline client/post-editor/media-modal/image-editor/image-editor-canvas.jsx]: Thanks, updated in 8aec671

File: client/post-editor/media-modal/index.jsx [hunk 3]
```diff
@@ -209,15 +209,15 @@ module.exports = React.createClass( {
 		this.setView( ModalViews.IMAGE_EDITOR );
 	},
 
-	onImageEditorClose: function() {
+	onImageEditorClose: function( event, { hasEditedImage } = { hasEditedImage: false } ) {
```
Comment (aduth) [inline client/post-editor/media-modal/index.jsx]: What will cause the second argument to be passed?

Comment (gwwar) [inline client/post-editor/media-modal/index.jsx]: I pass this in the onDone handler, https://github.com/Automattic/wp-calypso/pull/7374/files#diff-1f462f2e28f6f7a1f08bd085a3fca110R68

File: client/post-editor/media-modal/image-editor/index.jsx (hunk 2 above)
Comment (gwwar) [inline client/post-editor/media-modal/image-editor/index.jsx]: Oh good catch, updated in b64f86f

File: client/post-editor/media-modal/index.jsx (hunk 3 above)
Comment (aduth) [inline client/post-editor/media-modal/index.jsx]: A bit more concise, can drop the fallback empty object if second argument is always passed:

```js
{ hasEditedImage = false } = {}
```

Comment (aduth) [inline client/post-editor/media-modal/index.jsx]: >I pass this in the onDone handler,

Ah, thought I did, but I didn't have the branch checked out locally when looking at the file.

Comment (gwwar) [inline client/post-editor/media-modal/index.jsx]: Thinking on this more, it's probably cleaner to pass through a second handler: `onImageEditorCancel`

**Comment by aduth** [comment]
Took it for a spin and seems to work well in my testing

File: client/post-editor/media-modal/index.jsx (hunk 3 above)
Comment (gwwar) [inline client/post-editor/media-modal/index.jsx]: Updated to use 2 handlers in eaf28f1

**Comment by gwwar** [comment]
Thanks for 👀 @aduth!

**Comment by artpi** [comment]
Splendid ! WFM! 🚢  !

**Comment by gwwar** [comment]
Thanks for the review @artpi!

**Pull request merged**